### PR TITLE
dls-start-new-module: Add CSS template

### PR DESCRIPTION
**Pull request by hir12111**
A few things have been done:
- Add support for CookieCutter
- Add CSS template for the UI IOCs
- Add additional template arguments

Supporting CookieCutter has many advantages:
- It uses the template language jinja2 which is very flexible
- It allows defining pre-generation and post-generation hooks, therefore all the logic belonging to a template could be in the template folder

### CODE REVIEW

**Comment by ulrikpedersen** [comment]
Thanks @hir12111 that seems like a good idea to me.

Please, can @MJGaughran, @gilesknap and @GDYendell review and comment on here?

@hir12111 you will need to install cookie-cutter in the 'tools' area before we can release this... Also note that we have to be a little careful about versioning releases of dls_ade due to separate branches for RHEL6 and RHEL7.

File: dls_ade/cookiecutter_templates/CSS_IOC/{{cookiecutter.project_name}}/{{cookiecutter.domain}}App/Db/Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+DB += {{cookiecutter.domain}}-{{cookiecutter.technical_area}}-IOC-01.db
+
+#----------------------------------------------------
+# In a Diamond IOC Application, build xml files from
+# database files like this
+#
+
+# List database files required.
+USES_DB += ../O.Common/{{cookiecutter.domain}}-{{cookiecutter.technical_area}}-IOC-01.db
+# USES_DB += $(VA02)/db/BL21I-VA-IOC-02_expanded.db
```
Comment (GDYendell) [inline dls_ade/cookiecutter_templates/CSS_IOC/{{cookiecutter.project_name}}/{{cookiecutter.domain}}App/Db/Makefile]: Should this be here?

File: dls_ade/cookiecutter_templates/CSS_IOC/{{cookiecutter.project_name}}/{{cookiecutter.domain}}App/src/{{cookiecutter.domain}}-{{cookiecutter.technical_area}}-IOC-01Main.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* BL11I-MO-IOC-01Main.cpp */
```
Comment (GDYendell) [inline dls_ade/cookiecutter_templates/CSS_IOC/{{cookiecutter.project_name}}/{{cookiecutter.domain}}App/src/{{cookiecutter.domain}}-{{cookiecutter.technical_area}}-IOC-01Main.cpp]: Can this be templated.

Comment (hir12111) [reply dls_ade/cookiecutter_templates/CSS_IOC/{{cookiecutter.project_name}}/{{cookiecutter.domain}}App/src/{{cookiecutter.domain}}-{{cookiecutter.technical_area}}-IOC-01Main.cpp]: I missed that one, I'll change it, thanks

**Comment by hir12111** [comment]
Thank you for reviewing and sorry to put you out but I think that waiting for approval before merging is a good practice

**Pull request merged**